tests/unit/test_matrix.py

```diff
@@ -229,3 +229,16 @@ def and_its_blocks_know_their_internal_representation(self, request):
         blocks = _BaseSubtotals.blocks(fake_cube, fake_measure)
         _blocks.assert_called_once()
         assert blocks == fake_blocks
+
+    def it_knows_how_to_assemble_its_blocks(self, request):
+        bs = _BaseSubtotals(None, None)
+        bs._base_values = property_mock(request, _BaseSubtotals, "_base_values")
```

Comment:
You don't want to be assigning mocks to properties of the instance. The mocks already "do" that. I'm not sure why this is working :)

So you would have something like this instead. Note there is no need to name mocks if you don't refer to them later; flake8 will complain if you do :)

```
property_mock(request, _BaseSubtotals, "_base_values", return_value=[1])
property_mock(request, _BaseSubtotals, "_subtotal_columns", return_value=[2])
property_mock(request, _BaseSubtotals, "_subtotal_rows", return_value=[3])
property_mock(request, _BaseSubtotals, "_intersections", return_value=[4])

assert subtotals._blocks == [[[1], [2]], [[3], [4]]]
```

Reply:
yes, figured it out just before reading the comment... this was a first day blurp. fixed now...